frontend/server/controllers/ContestController.php

```diff
@@ -706,17 +710,35 @@ public static function apiDetails(Request $r) {
             } else {
                 $result['submission_deadline'] = min(
                     strtotime($r['contest']->finish_time),
-                    strtotime($problemset_user->access_time) + $r['contest']->window_length * 60
+                    strtotime($problemsetIdentity->end_time)
                 );
             }
             $result['admin'] = Authorization::isContestAdmin($r['current_identity_id'], $r['contest']);
 
-            // Log the operation.
-            ProblemsetAccessLogDAO::create(new ProblemsetAccessLog([
-                'identity_id' => $r['current_identity_id'],
-                'problemset_id' => $r['contest']->problemset_id,
-                'ip' => ip2long($_SERVER['REMOTE_ADDR']),
-            ]));
+            // Transaction begins
+            DAO::transBegin();
+
+            // Adding timer info separately as it depends on the current user and we don't
+            // want this to get generally cached for everybody
+            // Save the time of the first access
+            try {
+                // Log the operation.
+                ProblemsetAccessLogDAO::create(new ProblemsetAccessLog([
+                    'identity_id' => $r['current_identity_id'],
+                    'problemset_id' => $r['contest']->problemset_id,
+                    'ip' => ip2long($_SERVER['REMOTE_ADDR']),
+                ]));
+                DAO::transEnd();
+            } catch (InvalidParameterException $e) {
```

Comment:
```suggestion
            } catch (ApiException $e) {
```

así no necesitas el bloque en L735.

Reply:
Listo